src/Mod/Draft/drafttaskpanels/task_polararray.py

```diff
@@ -279,6 +284,64 @@ def get_center(self):
                             U.Quantity(c_z_str).Value)
         return center
 
+    def select_axis(self):
+        """Execute as callback when the axis button is clicked.
+
+        The axis selection process and button has the three stages
+        select (triggers selection), abort (aborts selection),
+        discard (set after axis is selected successfully and discards
+        the selected and saved axis)
+        """
+        if self.form.button_axis.text() == "select":
+            self.enable_axis_selection()
+            return
+        elif self.form.button_axis.text() == "abort" or self.form.button_axis.text() == "discard":
+            self.disable_axis_selection()
+
+    def display_axis(self, axis_name, edge_name, edge):
+        """Show the selected axis in the GUI. Add derived values for the center coordinates GUI."""
+        _msg("Selected: {} with edge {}".format(axis_name, edge_name))
+        self.axis_reference = "(Gui.ActiveDocument.getObject('{}').Object, " \
+                              "['{}'])".format(axis_name, edge_name)
```

Comment:
This is a bit convoluted, so that's why I suggest unpacking this selection and then using two variables.

Something like this
```py
self.axis_object = axis_name
self.axis_edge = edge_name
```
Then
```py
        _cmd += "center=" + DraftVecUtils.toString(self.center) + ", "
        _cmd += "axis_object=" + str(self.axis_object) + ", "
        _cmd += "axis_edge=" + str(self.axis_edge) + ", "
        _cmd += "use_link=" + str(self.use_link)
```